connector/connect/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectServiceSuite.scala

```diff
@@ -55,4 +65,38 @@ class SparkConnectServiceSuite extends SharedSparkSession {
           && schema.getFields(1).getType.getKindCase == proto.DataType.KindCase.STRING)
     }
   }
+
+  test("failures in the arrow collect path should not cause hangs") {
```

Comment:
nit
```suggestion
  test("SPARK-41165: failures in the arrow collect path should not cause hangs") {
```

logic-wise, this is what I would do too. Looks good to me.